### ISSUE TITLE
Add useResult

### ISSUE THREAD

**Issue by TimWhiting** [marcglasberg/fast_immutable_collections#24]
As of `meta: ^1.4.0` there is a new annotation called useResult that can be used to signal that a return value should not be discarded.

This will help users that are transitioning to immutable collections / learning how to use immutable collections.

For example a new user might think that:
```dart
final l = IList<int>();
l.add(0);
l.add(1);
print(l);
```
will print [0,1]
when actually it will print []

Adding this annotation will help newcomers on a project who aren't used to immutable collections or haven't read the documentation yet. You can even provide a reason statement:

```dart
@UseResult('You should always use the modified copy of the old list with the new item added')
IList<T> add(T item){
...
}
```

**Comment by marcglasberg** [comment]
The implementation of this is still buggy. It's saying things like `'replace' should be used.` in the comments!

**Comment by marcglasberg** [comment]
Fixed in version: 7.0.5-dev0
